index.tsx

```diff
@@ -0,0 +1,346 @@
+  }
+
+  backButtonHandler() {
+    if (this.props.visible) {
+      this.props.onCancel();
+      return true;
+    }
+    return false;
+  }
+  render() {
+    const { style = {}, overlayStyle = {}, visible, ...props } = this.props;
+    return (
+      <Overlay visible={visible} {...props}>
+        <View style={[styles.actionSheetContainer, overlayStyle]}>
+          <TouchableOpacity style={shareSheetStyle} onPress={this.props.onCancel} />
+          <Sheet visible={this.props.visible}>
+            <View style={[styles.buttonContainer, style]}>{this.props.children}</View>
+          </Sheet>
+        </View>
+      </Overlay>
+    );
+  }
+}
+
+type Options = {
+  url?: string;
+  urls?: Array<string>;
+  filename?: string;
+  filenames?: Array<string>;
+  type?: string;
+  message?: string;
+  title?: string;
+  subject?: string;
+  recipient?: string;
+  excludedActivityTypes?: string;
+  failOnCancel?: boolean;
+  showAppsToView?: boolean;
+  saveToFiles?: boolean;
+  appId: string;
+};
+type MultipleOptions = {
+  url?: string;
+  urls?: Array<string>;
+  filename?: string;
+  filenames?: Array<string>;
+  type?: string;
+  message?: string;
+  title?: string;
+  subject?: string;
+  activityItemSources?: Array<ActivityItemSource>;
+  excludedActivityTypes?: string;
+  failOnCancel?: boolean;
+  showAppsToView?: boolean;
+  saveToFiles?: boolean;
+};
+
+enum ActivityType {
+  addToReadingList = 'addToReadingList',
+  airDrop = 'airDrop',
+  assignToContact = 'assignToContact',
+  copyToPasteBoard = 'copyToPasteBoard',
+  mail = 'mail',
+  message = 'message',
+  openInIBooks = 'openInIBooks',
+  postToFacebook = 'postToFacebook',
+  postToFlickr = 'postToFlickr',
+  postToTencentWeibo = 'postToTencentWeibo',
+  postToTwitter = 'postToTwitter',
+  postToVimeo = 'postToVimeo',
+  postToWeibo = 'postToWeibo',
+  print = 'print',
+  saveToCameraRoll = 'saveToCameraRoll',
+  markupAsPDF = 'markupAsPDF',
+}
+interface IActivityType {
+  addToReadingList: string;
+  airDrop: string;
+  assignToContact: string;
+  copyToPasteBoard: string;
+  mail: string;
+  message: string;
+  openInIBooks: string;
+  postToFacebook: string;
+  postToFlickr: string;
+  postToTencentWeibo: string;
+  postToTwitter: string;
+  postToVimeo: string;
+  postToWeibo: string;
+  print: string;
+  saveToCameraRoll: string;
+  markupAsPDF: string;
+}
+
+type ActivityItem = { type: 'text' | 'url'; content: string };
+
+type LinkMetadata = {
+  originalUrl?: string;
+  url?: string;
+  title?: string;
+  icon?: string;
+  image?: string;
+  remoteVideoUrl?: string;
+  video?: string;
+};
+
+type ActivityItemSource = {
+  placeholderItem: ActivityItem;
+  item: { markupAsPDF: ActivityItem } | { [key: string]: ActivityItem };
+  subject?: Partial<IActivityType> | { [key: string]: string };
+  dataTypeIdentifier?: Partial<IActivityType> | { [key: string]: string };
+  thumbnailImage?: Partial<IActivityType> | { [key: string]: string };
```

Comment:
lmk about this typing here

Reply:
@Thomazella what about something like
```
subject?: { [key in ActivityType]?: string };
```
instead of 
```
subject?: Partial<IActivityType> | { [key: string]: string };
```
?

This way you don't need the `IActivityType` interface anymore :)

Reply:
Looks better, thanks! I have to finish this PR.